interactions/api/models/guild.py

```diff
@@ -2238,6 +2238,56 @@ async def create_emoji(
         self.emojis.append(_emoji)
         return _emoji
 
+    async def modify_emoji(
+        self,
+        emoji_id: Union[int, Snowflake, Emoji],
+        name: Optional[str] = MISSING,
+        roles: Optional[Union[List[Role], List[int]]] = MISSING,
+        reason: Optional[str] = None,
+    ) -> Emoji:
+        """
+        Edits an Emoji in the guild.
+
+        :param Union[int, Snowflake, Emoji] emoji_id: The id of the emoji to edit
+        :param Optional[str] name: The name of the emoji. If not specified, the filename will be used
+        :param Optional[Union[List[Role], List[int]]] roles: Roles allowed to use this emoji
+        :param Optional[str] reason: The reason of the modification
+        :return: The modified emoji object
+        :rtype: Emoji
+        """
+        if not self._client:
+            raise LibraryException(code=13)
+
+        if isinstance(emoji_id, Emoji):
+            emoji = emoji_id
+        else:
+            emoji = await self.get_emoji(int(emoji_id))
+
+        _name = emoji.name if name is MISSING else name
```

Comment:
No need to send original name of emoji.
Also you could get rid of `_name` variable like this
```py
if name is not MISSING:
    payload["name"] = name
```